### ISSUE TITLE
MD044: support Markdown inline links in HTML comments

### ISSUE THREAD

**Issue by eread** [rvben/rumdl#532]
### Problem or motivation

HTML comments are a common way to add additional context to Markdown. Sometimes you want to embed external links in comments, which can be wrapped in angle brackets and sometimes you want to embed relative links to refer to other Markdown files in a project.

It's possible to escape MD044 with a relative link by wrapping the reference to the file in backticks, but it might also be helpful if people could use Markdown inline links and `rumdl` could detect they're a link and shouldn't be treated as prose.

### Proposed solution

Have MD044 ignore text that is within a detected Markdown inline link so that given this configuration:

```toml
[MD044]
names = ["Test"]
```

This Markdown wouldn't be flagged with any violations:

```markdown
---
title: Level 1 heading
---

This is a Test.

<!-- For more information, see the [relevant page](test.md). -->
<!-- For more information, see `test.md` -->
```

At the moment, line 7 violates MD044.

### Feature category

Improve existing rule

### Example markdown

```markdown
---
title: Level 1 heading
---

This is a Test.

<!-- For more information, see the [relevant page](test.md). -->
<!-- For more information, see `test.md` -->
```

### Additional context

Links in HTML text like this don't become clickable Markdown links, but there doesn't seem to be any reason to not allow this style.

**Comment by rvben** [comment]
Fixed in v0.1.52. MD044 now skips proper name checks inside Markdown link URLs (`[text](url)`) and reference link labels (`[text][ref]`) within HTML comments, HTML blocks, and frontmatter. Link text is still checked as expected.

Thanks for reporting this @eread!
